xc/device.ml

```diff
@@ -1819,19 +1820,35 @@ module Dm_Common = struct
     ; fd_map = []
     }
 
-  let vgpu_args_of_nvidia domid vcpus (vgpu:Xenops_interface.Vgpu.nvidia) pci restore device =
+  let vgpu_args_of_nvidia domid vcpus vgpus restore =
     let open Xenops_interface.Vgpu in
+    let virtual_pci_address_compare vgpu1 vgpu2 =
+    match vgpu1, vgpu2 with
+    |{implementation = Nvidia nvi1},  {implementation = Nvidia nvi2} ->
+         begin
+             match nvi1.virtual_pci_address, nvi2.virtual_pci_address with
+             |Some pci1, Some pci2 -> Pervasives.compare pci1.dev pci2.dev
+             |_,_-> 0
```

Comment:
Is this correct? `0` means equal. So the function returns 0 in most cases. This does not look like it implements a total order.

Reply:
Ah, 0 means equal, it is wrong. 
I am going to combine the 2 matches and raise exception for incomparable  vGPUs as they are not expected here.